app/services/etherdelta_observer.py

```diff
@@ -41,14 +43,36 @@
 async def on_disconnect(io_client, event):
     logger.info("ED API client disconnected from %s", io_client.ws_url)
 
+
+
 async def process_orders(orders):
     not_deleted_filter = lambda order: "deleted" not in order or not order["deleted"]
     logger.info("Processing %i orders", len(orders))
     orders = list(filter(not_deleted_filter, orders))
     logger.debug("Filtered orders: %i", len(orders))
+    v = OrderMessageValidatorEtherdelta()
 
+    current_block = web3.eth.blockNumber # TODO: Introduce a strict timeout here; on failure allow order (todo copied from websocket_server.py)
     for order in orders:
         try:
+            if not v.validate(order):
+                error_msg = "Invalid message format"
+                details_dict = dict(data=order, errors=v.errors)
+                logger.warning("ED order rejected: %s: %s", error_msg, details_dict)
+                continue
+            else:
+                order_validated = v.document # Get data with validated and coerced values
+
+            if order_validated["expires"] <= current_block:
+                error_msg = "Cannot post order because it has already expired"
+                details_dict = { "blockNumber": current_block, "expires": order["expires"], "date": datetime.utcnow() }
+                logger.warning("ED Order rejected: %s: %s", error_msg, details_dict)
+                continue
+
+            if not order_signature_valid(order_validated):
+                logger.warning("ED Order rejected: Invalid signature: order = %s", order)
+                continue
```

Comment:
This feels like a candidate for a `filter` predicate function to me. What do you think?

Reply:
I think it's a matter of personal coding style. 
For me the separate validations inside the for loop are readable and easy to understand. I'm fine reading the same with a lambda function, but I tend to not write it like that. 
Go ahead and change it if you want.

Reply:
I don't think we can do it with a lambda; rather, just extract it into a function that returns False where we currently have `continue` and True at the end?

Reply:
I don't understand. Just go ahead and make the change.

Reply:
Check out 55ce3e6.